src/core/models/result.js

```diff
@@ -126,6 +129,63 @@ export default class Result {
       distance: result.distance,
       distanceFromFilter: result.distanceFromFilter,
       highlighted: highlightedData
+    };
+
+    const formattedData = this._getFormattedData(resultData, formatters, verticalKey);
+    resultData.formatted = formattedData;
+
+    if (formattedData.description !== undefined) {
+      resultData.details = formattedData.description;
+    }
+
+    return new Result(resultData);
+  }
+
+  /**
+   * Returns an object which contains formatted fields
+   *
+   * @param {Object<string, function>} formatters to apply to the result fields
+   * @param {string} verticalKey the verticalKey associated with the result
+   * @returns {Object<string, string>} keys are field names and values are the formatted data
+   */
+  static _getFormattedData (resultData, formatters, verticalKey) {
```

Comment:
Looking through the SDK code, it seems like we only formatted data if it came from the KnowledgeManager back-end. Do we need to have such a safeguard here?

Reply:
That's a good catch. Since we recently added a Source param to the result, I can adda. check to make sure that the result is from KnowledgeManager before applying formatting